src/prefect/blocks/core.py

```diff
@@ -629,7 +629,7 @@ def _generate_code_example(cls) -> str:
         """Generates a default code example for the current class"""
         qualified_name = to_qualified_name(cls)
         module_str = ".".join(qualified_name.split(".")[:-1])
-        class_name = cls.__name__
+        class_name = cls.__name__.partition("[")[0]  # Remove generic parameters
```

Comment:
this is to make the block standards tests pass, i.e. not display `from prefect.blocks.system import Secret[str]`

Reply:
I suspect https://docs.python.org/3/library/types.html#types.get_original_bases would help here (it's new in 3.12 but is easily back ported).

Reply:
Actually, on second thought: it isn't going to help here. Maybe `typing.get_origin()` would but it's late here so all you get is a vague pointer for now. 🫠

Reply:
As the 99 captain might say: "Bingpot!":

```python
>>> Secret[str].__pydantic_generic_metadata__
{'origin': <class 'prefect.blocks.system.Secret'>, 'args': (<class 'str'>,), 'parameters': ()}
```

Documented at https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel

Reply:
nice find! good to know - updated